Emoji plugin: Refactored and added logic for Dropdown

bojangles-m opened the pull request:
Refactored Emoji plugin to add business logic for Dropdown plugin. 
It has no view and it is only added logic to keep PR smaller.
The Flyout index search and Flyout library is added which are used to search and show emojis in Dropdown.
useEmojiPicker is added to control the Dropdown and actions over it.
The view will be added in separate PR.

Small peek inside :)
https://www.loom.com/share/d7959dc612484aad8ac96c7af7f94a7b

zbeyens commented on packages/nodes/emoji/src/constants.ts:
```diff
@@ -1,5 +1,55 @@
+
 export const ELEMENT_EMOJI = 'emoji';
-export const ELEMENT_EMOJI_INPUT = 'emoji_input';
+// export const ELEMENT_EMOJI_INPUT = 'emoji_input';
```
can be removed

bojangles-m replied:
Removed.

zbeyens reviewed (commented):
Great PR!

What is a "flyout"? There's maybe already a naming convention for that in Plate.

Also, should we remove:
```
  isElement: true,
  isInline: true,
  isVoid: true,
```
and rename `ELEMENT_EMOJI` to `KEY_EMOJI` as it's not an element but just text?

bojangles-m commented:
> Great PR!
> 
> What is a "flyout"? There's maybe already a naming convention for that in Plate.
> 
> Also, should we remove:
> 
> ```
>   isElement: true,
>   isInline: true,
>   isVoid: true,
> ```
> 
> and rename `ELEMENT_EMOJI` to `KEY_EMOJI` as it's not an element but just text?

Updated the code it.

Flyout is the association for modal appearing when pressing on the emoji toolbar button. I saw it in the code and used it. Should I replaced it? Any suggestions?

zbeyens commented:
@bojangles-m We're using "Floating" naming for popovers, so we can use it for that case too

zbeyens commented:
@bojangles-m let's fix the build then I'll release this asap

bojangles-m commented:
> @bojangles-m let's fix the build then I'll release this asap

What is happening with MarkBalloonToolbar
It is breaking at line 32
```
const boldTooltip: TippyProps = { content: 'Bold (⌘+B)', ...markTooltip };
  const italicTooltip: TippyProps = { content: 'Italic (⌘+I)', ...markTooltip };
  const underlineTooltip: TippyProps = {
    content: 'Underline (⌘+U)',
    ...markTooltip,
  };
```
